feat(DragDrop): new drag drop styling fixes (#9784)
patternfly/patternfly-react: packages/react-core/src/components/DataList/examples/DataList.md

    'DataListItemRow',
    'DataListToggle',
    'DataListContent',
    'DataListDragButton',

>     'DataListControl'

Reviewer: just double checking - we could remove this, but we could either:
- additionally link to the new dragdrop docs to show the new recommended way to implement the draggable datalist (since that example now lives in DragDrop rather than datalist)
- OR keep these examples here, mark them deprecated, and possibly still link to the new recommended solution in the DragDrop examples.

Reviewer: The same would be true for the dual list selector

Author: Added back the examples as the old dragDrop is still available, and added a note pointing to the new implementation.

### Draggable

Note: There is a new recommended drag and drop implementation that has full keyboard functionality. Examples and details may be located [here](/components/drag-and-drop/react-next).

> Draggable data lists used to have their own HTML5-based API for drag and drop, which wasn't able to fulfill requirements such as custom styling on items being dragged. So we wrote generic `DragDrop`, `Draggable`, and `Droppable` components for this purpose. Use those new components instead of the deprecated (and buggy!) HTML5-based API. Keyboard accessibility and screen reader accessibility for the `DragDrop` component are still in development.

Reviewer 2: ```suggestion
Previously, draggable data lists had their own API for the [drag and drop component](/components/drag-and-drop), which wasn't able to fulfill requirements like custom styling for items as they are dragged. 

To address this disparity, `<DragDrop>`, `<Draggable>`, and `<Droppable>` components were added to replace our now deprecated HTML5-based API. 

Note that keyboard and screen reader accessibility functionality for the `<DragDrop>` component are still in development.
```

I'm inclined to reword the "fulfill requirements" part of the first sentence, because I'm not quite understanding what is meant by that, but I can't tell what's important to convey. What kind of requirements is this referring to - accessibility? Were there others besides "custom styling" that could help buff out this sentence? The way it reads now feels incomplete, but maybe it's just because I'm unfamiliar with the previous implementation

Author: I think it's referring to general flexibility of the implementation, and "requirements" in the sense of what a user might want or need. 

`Previously, draggable data lists had their own API for the [drag and drop component](/components/drag-and-drop), which wasn't flexible enough to allow custom styling for items as they are dragged.` Does that sound better?

Reviewer 2: Yeah that sounds awesome! Thanks for clarifying

### Draggable

> Note: There is a new recommended drag and drop implementation with full keyboard functionality, which replaces this implementation. To adhere to our new recommendations, refer to these related [drag and drop examples](/components/drag-and-drop/react-next).

Reviewer 2: ```suggestion
Note: This implementation will eventually be replaced by a new drag and drop implementation, with full keyboard functionality. To adhere to this new recommendation, refer to the [drag and drop demos](components/drag-and-drop/react-next-demos).
```
wdyt about this phrasing?

Previously, draggable data lists had their own API for the [drag and drop component](/components/drag-and-drop), which wasn't flexible enough to allow custom styling for items as they are dragged. To address this disparity, `<DragDrop>`, `<Draggable>`, and `<Droppable>` components were added to replace our now deprecated HTML5-based API.

> Note that keyboard and screen reader accessibility functionality for the `<DragDrop>` component are still in development.

Reviewer 2: ```suggestion
Keyboard and screen reader accessibility for the `<DragDrop>` component is still in development.
```

### Draggable

> Note: There is a new recommended drag and drop implementation with full keyboard functionality, which replaces this implementation. To adhere to our new recommendations, refer to the [drag and drop demos](components/drag-and-drop/react-next-demos).

Reviewer 2: ```suggestion
Note: There is a new recommended drag and drop implementation with full keyboard functionality, which replaces this implementation. To adhere to our new recommendations, refer to the [drag and drop demos](/components/drag-and-drop/react-next-demos).
```

> Note: There is a new recommended drag and drop implementation with full keyboard functionality, which replaces this implementation. To adhere to our new recommendations, refer to these related [drag and drop examples](/components/drag-and-drop/react-next).

Reviewer 2: ```suggestion
Note: There is a new recommended drag and drop implementation with full keyboard functionality, which replaces this implementation. To adhere to our new recommendations, refer to the [drag and drop demos](components/drag-and-drop/react-next-demos).
```
